### PR TITLE
Update Sonarqube Mapper to include branch name in calls 

### PR DESCRIPTION
**Pull request by HenryXiaoHX**
- [x] Manually tested by Amndeep using MITRE Sonarqube instance.
Signed-off-by: HenryXiaoHX <hxiao@ucsd.edu>

### CODE REVIEW

**Review by Amndeep7 — APPROVED** [review]

File: libs/hdf-converters/src/sonarqube-mapper.ts [hunk 1]
```diff
@@ -146,54 +146,50 @@ export class SonarQubeResults {
...
-        });
-      })
+    await Promise.all(
+      this.data.issues?.map((issue) =>
+        axios
+          .get(`${this.sonarQubeHost}/api/sources/raw`, {
+            auth: {username: this.userToken, password: ''},
+            params: {
+              key: issue.component,
+              ...(this.branchName && {branch: this.branchName})
```
Comment (em-c-rod) [inline libs/hdf-converters/src/sonarqube-mapper.ts:158]: What is the result of having logic within what is operated on by the spread operator `...`?

Comment (em-c-rod) [reply libs/hdf-converters/src/sonarqube-mapper.ts:158]: Also, what is the purpose of adding in the branch name? Where does it get used?

Comment (em-c-rod) [reply libs/hdf-converters/src/sonarqube-mapper.ts:158]: Understood :+1:

**Review by em-c-rod — APPROVED** [review]
What is the pattern for testing it next time? Just food for thought for the team.

**Comment by Amndeep7** [comment]
@mergifyio refresh

**Pull request merged**